app/src/parser.c

```diff
@@ -188,52 +215,87 @@ parser_error_t parser_getItem(const parser_context_t *ctx,
                                    ctx->tx_obj->callIndex.moduleIdx, ctx->tx_obj->callIndex.idx, argIdx,
                                    outVal, outValLen,
                                    pageIdx, pageCount);
+        return err;
     } else {
         // CONTINUE WITH FIXED ARGUMENTS
-        // FIXME: Review. We can probably show this only in expert mode
         displayIdx -= methodArgCount;
-        switch (displayIdx) {
-            case FIELD_NETWORK:
-                if (_getAddressType() == PK_ADDRESS_TYPE) {
-                    snprintf(outKey, outKeyLen, "Chain");
-                    snprintf(outVal, outValLen, COIN_NAME);
-                    break;
-                }
+        if( displayIdx == FIELD_NETWORK ){
+            if (_getAddressType() == PK_ADDRESS_TYPE) {
+                snprintf(outKey, outKeyLen, "Chain");
+                snprintf(outVal, outValLen, COIN_NAME);
+            }else {
                 snprintf(outKey, outKeyLen, "Genesis Hash");
                 _toStringHash(&ctx->tx_obj->genesisHash,
                               outVal, outValLen,
                               pageIdx, pageCount);
-                break;
-            case FIELD_ONCE:
+            }
+            return err;
+        }
+
+        if( displayIdx == FIELD_NONCE ) {
+            if(parser_show_expert_fields()){
                 snprintf(outKey, outKeyLen, "Nonce");
                 _toStringCompactIndex(&ctx->tx_obj->nonce,
                                       outVal, outValLen,
                                       pageIdx, pageCount);
-                break;
-            case FIELD_TIP:
+                return err;
+            }
+            displayIdx++;
+        } else if( !parser_show_expert_fields() ){
+            displayIdx++;
+        }
```

Comment:
Two comments here:
- we are not returning the error that _toStringCompactIndex returns
- maybe we can simplify a bit the logic to avoid nested ifs?

```
        if( parser_show_expert_fields() && displayIdx == FIELD_NONCE ) {
            snprintf(outKey, outKeyLen, "Nonce");
            return _toStringCompactIndex(&ctx->tx_obj->nonce,
                                  outVal, outValLen,
                                  pageIdx, pageCount);
        }

        if( !parser_show_expert_fields() ) {
            displayIdx++;
        }
```

we can probably apply similar logic to the other cases...

Reply:
yes, you are right. I checked and it works